### PR TITLE
8273635: Attempting to acquire lock StackWatermark_lock/9 out of order with lock tty_lock/3

### PR DESCRIPTION
**Pull request by coleenp**
This change reverts the rank ordering of ttyLock and StackWatermark_lock because the latter is held through a very large region and printing all of this to a buffer with xmlstream is non-trivial.
With this change, if tty->print_cr() is done while holding the stackwatermark lock or lower (which is service ranking, etc), a lock inversion will happen with ttyLock.  This doesn't happen now because all the code in GC and much of the rest of the runtime use UL and not tty->print().
Tested with tier1-6.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273635](https://bugs.openjdk.java.net/browse/JDK-8273635): Attempting to acquire lock StackWatermark_lock/9 out of order with lock tty_lock/3

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 53dd04d23b3fdedf04be2071c4a35bfb0943d9c8
 * [Erik Österlund](https://openjdk.java.net/census#eosterlund) (@fisk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5499/head:pull/5499` \
`$ git checkout pull/5499`

Update a local copy of the PR: \
`$ git checkout pull/5499` \
`$ git pull https://git.openjdk.java.net/jdk pull/5499/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5499`

View PR using the GUI difftool: \
`$ git pr show -t 5499`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5499.diff">https://git.openjdk.java.net/jdk/pull/5499.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/compiler/uncommontrap/TestDeoptOOM.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 /*
  * @test
  * @bug 8273456
- * @summary Test that ttyLock isn't held when taking StackWatermark_lock
+ * @summary Test that ttyLock is ranked about StackWatermark_lock
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/compiler/uncommontrap/TestDeoptOOM.java:39]: s/about/above/ ?

**Review by dholmes-ora — APPROVED** [review]
Hi Coleen,

The ranking restoration seems fine. One possible typo below.

Thanks,
David

File: test/hotspot/jtreg/compiler/uncommontrap/TestDeoptOOM.java (hunk 1 above)
Comment (coleenp) [reply test/hotspot/jtreg/compiler/uncommontrap/TestDeoptOOM.java]: fixed, thanks!

**Comment by coleenp** [comment]
Thanks for the review, David!

**Review by fisk — APPROVED** [review]
The rank reversal looks good. Too bad there was more printing code.

**Comment by coleenp** [comment]
Thanks Erik!
/integrate

**Pull request closed without merging**